### PR TITLE
Complain when file does not exist

### LINKED ISSUE #522: join-contracts.py should complain about  missing imports
Currently, `join-contracts.py` succeeds silently when some of the imported files are not found. This issue keeps track of making `join-contracts.py` fail for missing imports.

#### Current behavior
`TokenNetwork.sol` imports many contracts, but it gets smaller after `join-contracts.py` processing.

```
$ utils/join-contracts.py contracts/TokenNetwork.sol flat.sol && wc -l contracts/TokenNetwork.sol flat.sol
Reading contracts/TokenNetwork.sol
  1684 contracts/TokenNetwork.sol
  1679 flat.sol
  3363 total
```

### PR DESCRIPTION
**Pull request by pirapira**
### What this PR does

`join-contracts.py` raises an exception when a file does not exist.

### Why I'm making this PR

Before this commit, join-contracts.py silently skipped
non existing files.  The result is usually not useful
and might lead to nonpleasant surprises.

This closes #522.

### What's tricky about this PR (if any)

none.
----

Any reviewer can check these:

* [ ] If the PR is fixing a bug or adding a feature, add an entry to CHANGELOG.md.
* [ ] If the PR changed a Solidity source, run `make compile_contracts` and add the resulting `raiden_contracts/data/contracts.json` in the PR.
* [ ] If the PR is changing documentation only, add `[skip ci]` in the commit message so Travis does not waste time.
    * [ ] But, if the PR changes comments in a Solidity source, do not add `[skip ci]` and let Travis check the hash of the source.
* [ ] In Python, use keyword arguments
* [ ] Squash unnecessary commits
* [ ] Comment commits
* [ ] Follow naming conventions
    * `solidityFunction`
    * `_solidity_argument`
    * `solidity_variable`
    * `python_variable`
    * `PYTHON_CONSTANT`
* [ ] Follow the Signature Convention in [CONTRIBUTING.md](./CONTRIBUTING.md)
* For each new contract
    * [ ] The deployment script deploys the new contract.
    * [ ] `etherscan_verify.py` runs on the new contract.
* Bookkeep
    * [ ] The gas cost of new functions are stored in `gas.json`.
* Solidity specific conventions
    * [ ] Document arguments of functions in natspec
    * [ ] Care reentrancy problems
* [ ] When you catch a require() failure in Solidity, look for a specific error message like `pytest.raises(TransactionFailed, match="error message"):`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

### CODE REVIEW

**Comment by pirapira** [comment]
CodeCov complains.  It's a good time to add tests.

File: raiden_contracts/tests/test_contract_joiner.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            joiner.join(open(source_file.name))
+
+
+def test_contract_joiner_with_empty_file() -> None:
+    """ Using ContractJoiner on an empty source """
+    joiner = ContractJoiner()
+    empty_source = ""
+    with NamedTemporaryFile() as source_file:
+        source_file.write(bytearray(empty_source, "ascii"))
+        source_file.flush()
```
Comment (karlb) [inline raiden_contracts/tests/test_contract_joiner.py]: You don't need to explicitly write nothing to the source file. It is already empty before.

Comment (pirapira) [reply raiden_contracts/tests/test_contract_joiner.py]: Copy-paste-modify did harm here.  I'll remove these lines and the empty string.

File: raiden_contracts/utils/join_contracts.py [hunk 2]
```diff
@@ -61,9 +61,10 @@ def _on_import_line(self, stripped_line: str, out: List[str]) -> None:
             for prefix, path in self.import_map.items():
                 if next_file.startswith(prefix):
                     next_file = next_file.replace(prefix, path)
-            if next_file and os.path.exists(next_file):
-                with open(next_file) as next_contract:
-                    out.extend(self.join(next_contract))
+            if not os.path.exists(next_file):
+                raise FileNotFoundError(f"File does not exist: {next_file}")
```
[the remark this answers is not in the document]
Comment (pirapira) [reply raiden_contracts/utils/join_contracts.py]: I'll remove these lines.

**Pull request merged**